### ISSUE TITLE
Fixes seem a little too "aggressive"

### ISSUE THREAD

**Issue by serl** [rvben/rumdl#69]
Hello, trying to slide this into all pre-commits configuration under my control 😁 

Let's say that our markdowns are not the best formatted to start with, and I had a little problem. While stumbling on #67, half of the README went in flames. I agree that's probably outdated and to rewrite but for now I only wanted to install a linter! 🤣 

Jokes apart, I managed to produce a compact example (probably not the minimal one, but it's a start...) that shows the behavior:

```md
# Test file

## Blabla

Blabla

### First

Blabla.

- [the-link]: blablabla

Blablabla bla.

[the-link]: https://example.com/

### Everything in the middle will disappear

This paragraph for example.

## Any other paragraph for that matter

There seems to be a list with a sub-item to make this explode:

- list
  - sub-item

This line will stay.

## Last paragraph

This will stay as well
```

`rumdl fmt` (v0.0.128) says:

```text
[MD053] Unused link/image reference: [the-link] [fixed]
```

And transforms the file to:
```md
# Test file

## Blabla

Blabla

### First

Blabla.

- [the-link]: blablabla

Blablabla bla.

This line will stay.

## Last paragraph

This will stay as well
```

That is:
- Removing lines from `[the-link]: https://example.com/`, to `  - sub-item`
- (not shown above) Removing the newline at the end of the file

Expected result is that the file is valid of course, and that's #67. But the extra lines removed are maybe an extra bug? 🤔

**Comment by rvben** [comment]
Hi @serl,

Thanks for raising this issue. I reconsidered if we can even fix MD053 safely, and decided to not provide an auto-fix for now. If there's a problem, it probably needs a (manual) fix instead of just removing it.

Do you agree?

**Comment by serl** [comment]
Yep I think that'll work, thanks for the very fast response!
